portal/views/auth.py

```diff
@@ -361,9 +362,30 @@ def base64_url_decode(s):
 def flask_user_login_event(app, user, **extra):
     auditable_event("local user login", user_id=user.id, subject_id=user.id,
                     context='login')
+
+    # After a successfull login make sure lockout is reset
+    user.reset_lockout()
+
     login_user(user, 'password_authenticated')
 
 
+def flask_user_password_failed_event(app, user, **extra):
+    """tracks when a user fails password verification
+
+    If this happens too often, for security reasons,
+    the user will be locked out of the system.
+    """
+    count = user.add_password_verification_failure()
+    auditable_event(
```

Comment:
>Should this be a normal log instead of an auditable event because the user is not authenticated?

Good point, that might make more sense

>If I should be a normal log, is it safe to use the user's id in the log?

Yup yup, those aren't publicly visible anywhere